crates/sui-proxy/src/admin.rs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::config::{PeerValidationConfig, RemoteWriteConfig};
+use crate::handlers::publish_metrics;
+use crate::middleware::{expect_mysten_proxy_header, expect_valid_public_key};
+use crate::peers::SuiNodeProvider;
+use anyhow::Result;
+
+use axum::routing::post as axum_post;
+use axum::Extension;
+use axum::{middleware, Router};
+use fastcrypto::ed25519::{Ed25519KeyPair, Ed25519PublicKey};
+use fastcrypto::traits::KeyPair;
+use std::fs;
+use std::io::BufReader;
+use std::net::SocketAddr;
+use std::sync::Arc;
+use std::time::Duration;
+
+use sui_tls::{rustls::ServerConfig, AllowAll, CertVerifier, SelfSignedCertificate, TlsAcceptor};
+use tokio::signal;
+
+use tower::ServiceBuilder;
+use tower_http::{
+    trace::{DefaultOnResponse, TraceLayer},
+    LatencyUnit,
+};
+use tracing::{info, Level};
+
+/// user agent we use when posting to mimir
+static APP_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
+
+/// Configure our graceful shutdown scenarios
+pub async fn shutdown_signal(h: axum_server::Handle) {
+    let ctrl_c = async {
+        signal::ctrl_c()
+            .await
+            .expect("failed to install Ctrl+C handler");
+    };
+
+    #[cfg(unix)]
+    let terminate = async {
+        signal::unix::signal(signal::unix::SignalKind::terminate())
+            .expect("failed to install signal handler")
+            .recv()
+            .await;
+    };
+
+    #[cfg(not(unix))]
+    let terminate = std::future::pending::<()>();
+
+    tokio::select! {
+        _ = ctrl_c => {},
+        _ = terminate => {},
+    }
+
+    let grace = 30;
+    info!(
+        "signal received, starting graceful shutdown, grace period {} seconds, if needed",
+        &grace
+    );
+    h.graceful_shutdown(Some(Duration::from_secs(grace)))
+}
+
+/// Reqwest client holds the global client for remote_push api calls
+/// it also holds the username and password.  The client has an underlying
+/// connection pool.  See reqwest documentation for details
+#[derive(Clone)]
+pub struct ReqwestClient {
+    pub client: reqwest::Client,
+    pub settings: RemoteWriteConfig,
+}
+
+pub fn make_reqwest_client(settings: RemoteWriteConfig) -> ReqwestClient {
+    ReqwestClient {
+        client: reqwest::Client::builder()
+            .user_agent(APP_USER_AGENT)
+            .pool_max_idle_per_host(32)
```

Comment:
yes, we definitely should.  I added the 32 value here because the default value is `std::usize::MAX`, which is massively larger 🔥 The configurable runtime configs really should be more akin to something that the service subscribes to and we can programmatically control and canary values to.  There are many details there though, like builders that would need to be recreated, etc.  I'm happy to add this to the config though.  I'll go ahead and do that with a default there of 32 as well.